crates/accelerate/src/two_qubit_decompose.rs

```diff
@@ -2344,6 +2344,352 @@ pub fn local_equivalence(weyl: PyReadonlyArray1<f64>) -> PyResult<[f64; 3]> {
     Ok([g0_equiv + 0., g1_equiv + 0., g2_equiv + 0.])
 }
 
+/// invert 1q gate sequence
+fn invert_1q_gate(gate: (StandardGate, SmallVec<[f64; 3]>)) -> (StandardGate, SmallVec<[f64; 3]>) {
+    let gate_params = gate.1.into_iter().map(Param::Float).collect::<Vec<_>>();
+    let Some(inv_gate) = gate
+        .0
+        .inverse(&gate_params) else {panic!()};
+    let inv_gate_params = inv_gate
+        .1
+        .into_iter()
+        .map(|param| match param {
+            Param::Float(val) => val,
+            _ => panic!(),
+        })
+        .collect::<SmallVec<_>>();
+    (inv_gate.0, inv_gate_params)
+}
+
+/// invert 2q gate sequence
+fn invert_2q_gate(
+    gate: (Option<StandardGate>, SmallVec<[f64; 3]>, SmallVec<[u8; 2]>),
+) -> (StandardGate, SmallVec<[f64; 3]>, SmallVec<[u8; 2]>) {
+    let Some(inv_gate) = gate
+                    .0
+                    .unwrap()
+                    .inverse(&gate.1.into_iter().map(Param::Float).collect::<Vec<_>>()) else {panic!()};
+    let inv_gate_params = inv_gate
+        .1
+        .into_iter()
+        .map(|param| match param {
+            Param::Float(val) => val,
+            _ => panic!(),
+        })
+        .collect::<SmallVec<_>>();
+    (inv_gate.0, inv_gate_params, gate.2)
+}
+
+#[pyclass(module = "qiskit._accelerate.two_qubit_decompose", subclass)]
+pub struct TwoQubitControlledUDecomposer {
+    #[pyo3(get)]
+    rxx_equivalent_gate: StandardGate,
```

Comment:
Currently, `rxx_equivalent_gate` could be only a `StandardGate` since in the code we need to invert it, and there is currently no inverse function for other types, see these lines:
```
                let circ_c = self.to_rxx_gate(gamma)?;
...
                for gate in circ_c.gates.into_iter().rev() {
                    let (inv_gate_name, inv_gate_params, inv_gate_qubits) = invert_2q_gate(gate);
```

Reply:
As for making this class public, I think that it may be useful. Should we do it now or only after it's added to the `UnitarySynthesis` transpiler pass #13320?

Reply:
I'd say lets do it later as the PR is large and intricate enough that we should review the change making it public on it's own.